packages/appstore/src/components/currency-switcher-card/demo/demo-account-card.tsx

```diff
@@ -3,19 +3,34 @@ import { observer } from 'mobx-react-lite';
 import { Button, Text } from '@deriv/components';
 import CurrencySwitcherContainer from 'Components/containers/currency-switcher-container';
 import BalanceText from 'Components/elements/text/balance-text';
-import { useStores } from 'Stores/index';
 import './demo-account-card.scss';
 import { localize } from '@deriv/translations';
+import {
+    useCFDDemoAccounts,
+    useCurrencyExchangeRate,
+    usePlatformDemoAccount,
+    useTotalAccountBalance,
+} from '@deriv/hooks';
+import { useStore } from '@deriv/stores';
 
 const DemoAccountCard = () => {
-    const { client, traders_hub } = useStores();
-    const { accounts, loginid, resetVirtualBalance } = client;
-    const { platform_demo_balance, selected_account_type } = traders_hub;
+    const { client, traders_hub } = useStore();
+    const { accounts, loginid, resetVirtualBalance, default_currency } = client;
+    const { selected_account_type } = traders_hub;
+    const platform_demo_account = usePlatformDemoAccount();
+    const cfd_demo_accounts = useCFDDemoAccounts();
+    const cfd_demo_balance = useTotalAccountBalance(cfd_demo_accounts);
+    const cfd_demo_rate = useCurrencyExchangeRate(platform_demo_account?.currency || default_currency);
 
     const canResetBalance = () => {
-        return accounts[loginid]?.balance !== 10000;
+        return loginid && accounts[loginid]?.balance !== 10000;
     };
 
+    const DemoAccountTotalBalance = React.useMemo(
+        () => (platform_demo_account?.balance || 0) + cfd_demo_balance.balance * cfd_demo_rate,
+        [cfd_demo_balance.balance, cfd_demo_rate, platform_demo_account?.balance]
+    );
```

Comment:
```suggestion
    const demo_account_total_balance = (platform_demo_account?.balance || 0) + cfd_demo_balance.balance * cfd_demo_rate;
```
@shayan-deriv We can simply use derived state and let react handle things.